### PR TITLE
Fix module links in std::fmt and the Rust book's documentation chapter.

### PR DESCRIPTION
**Pull request by luser**
The links in the rustdoc for several places in fmt were trying to link to
the std::fmt module but actually linking to std, which was confusing.
While trying to figure out why I noticed that the documentation chapter of
the Rust book has examples that show this same bug (although it doesn't seem
widespread in practice).

r? @steveklabnik

### CODE REVIEW

**Comment by luser** [comment]
FYI I grepped for other instances of this pattern and spot-checked the rest and they were all correct:
```
find . \( -name "*.rs" -o -name "*.md" \) -print0 | xargs -0 egrep "module[- ]level documentation"
```

File: src/libcore/fmt/mod.rs [hunk 1]
```diff
@@ -298,7 +298,7 @@ impl<'a> Display for Arguments<'a> {
 ///
 /// For more information on formatters, see [the module-level documentation][module].
```
Comment (Manishearth) [inline src/libcore/fmt/mod.rs]: We actually _don't_ have any info here either. http://doc.rust-lang.org/nightly/core/fmt/index.html

Should link to `../../std/fmt/index.html`, perhaps?

**Comment by Manishearth** [comment]
r+ with that fix

File: src/libcore/fmt/mod.rs (hunk 1 above)
Comment (luser) [inline src/libcore/fmt/mod.rs]: Ah, I was looking at http://doc.rust-lang.org/nightly/std/fmt/trait.Display.html, which with my patch winds up linking to http://doc.rust-lang.org/nightly/std/fmt/index.html. I didn't realize the docs would wind up in two places.

Comment (Manishearth) [inline src/libcore/fmt/mod.rs]: Yeah, `libstd` is mostly reexports of other, possibly unstable crates (collections, core, alloc).

**Comment by Manishearth** [comment]
@bors r+

Thanks!

**Comment by steveklabnik** [comment]
Thanks so much for this!

**Pull request merged**